src/components/MovieListPage/MovieListPage.tsx

```diff
@@ -74,13 +61,25 @@ export function MovieListPage() {
   });
 
   useEffect(() => {
+    const query = searchParams.get('query');
```

Comment:
Lets move it outside useEffect and use for both effects and update dependency arrays